### PR TITLE
Better incorrect response handling

### PR DESCRIPTION
**Pull request by maxme**
To test:

Checkout 41b236d and run the tests, check 1 test is failing (due to ` NumberFormatException`) - checkout fad4503 and check the test is fixed.

Note this doesn't fxi #134 (need more work and tests to have bullet proof clients) but is required for sitestore integration.

### CODE REVIEW

**Comment by maxme** [comment]
Fix #80

File: example/src/test/java/org/wordpress/android/fluxc/site/SiteXMLRPCClientTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    @Before
+    public void setUp() {
+        ShadowLog.stream = System.out;
+
+        mMockedQueue = mock(RequestQueue.class);
+        when(mMockedQueue.add(any(Request.class))).thenAnswer(new Answer<Void>() {
+            public Void answer(InvocationOnMock invocation) {
+                XMLRPCRequest request = (XMLRPCRequest) invocation.getArguments()[0];
+                try {
+                    Class<XMLRPCRequest> requestClass = (Class<XMLRPCRequest>) Class.forName("org.wordpress.android.fluxc.network.xmlrpc.XMLRPCRequest");
```
Comment (aforcier) [inline example/src/test/java/org/wordpress/android/fluxc/site/SiteXMLRPCClientTest.java]: This line is over the 120 character style limit

File: example/src/test/java/org/wordpress/android/fluxc/site/SiteXMLRPCClientTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                    // Object o = request.parseNetworkResponse(data)
+                    Method parseNetworkResponse = requestClass.getDeclaredMethod("parseNetworkResponse", NetworkResponse.class);
+                    parseNetworkResponse.setAccessible(true);
+                    NetworkResponse nr = new NetworkResponse(mMockedResponse.getBytes());
+                    Response<Object> o = (Response<Object>) parseNetworkResponse.invoke(request, nr);
+                    // request.deliverResponse(Object o)
```
Comment (aforcier) [inline example/src/test/java/org/wordpress/android/fluxc/site/SiteXMLRPCClientTest.java]: Couple of stale comments left behind in this method

Comment (maxme) [inline example/src/test/java/org/wordpress/android/fluxc/site/SiteXMLRPCClientTest.java]: Hmm, that's the actual call, we should keep it, the reflection code is pretty hard to read IMO.

**Comment by aforcier** [comment]
Looks good, tests are passing and I can log into my non-JP self-hosted site in the example app without crashing 😂

:shipit:

**Pull request merged**